node/lib/util/diff_util.js

```diff
@@ -129,41 +129,38 @@ exports.getRepoStatus = co.wrap(function *(repo,
         options.flags = options.flags |
                         NodeGit.Diff.OPTION.RECURSE_UNTRACKED_DIRS;
     }
-    const indexDiff =
-               yield NodeGit.Diff.treeToIndex(repo, tree, null, options);
-    const indexStatus = readDiff(indexDiff);
     const index = yield repo.index();
     const workdirToIndexDiff =
                        yield NodeGit.Diff.indexToWorkdir(repo, index, options);
     const workdirToIndexStatus = readDiff(workdirToIndexDiff);
     if (!workdirToTree) {
+        const indexToTreeDiff =
+            yield NodeGit.Diff.treeToIndex(repo, tree, null, options);
+        const indexToTreeStatus = readDiff(indexToTreeDiff);
         return {
-            staged: indexStatus,
+            staged: indexToTreeStatus,
             workdir: workdirToIndexStatus,
         };
     }
-    const workdirDiff =
```

Comment:
The alternate situation is used to derive the diff used when, for example, the user passes `-a` to `commit`, e.g.:

```
$ touch foo # new file
$ git add foo
$ rm foo
$ git status
On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        new file:   foo

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    foo
```

Yet:

```
$ git commit -a
On branch master
nothing to commit, working tree clean
```

Similarly if a file been changed to one state in the index, and back to the original (HEAD) state in the workdir.